### PR TITLE
Add REST API endpoint to read/update general settings 

### PR DESCRIPTION
**Pull request by Shelob9**
#2694 
Endpoints added via this PR:

* `POST /wp-json/cf-api/v2/settings` - Update general settings.
* `GET /wp-json/cf-api/v2/settings` - Read general settings.

### CODE REVIEW

**Comment by Shelob9** [comment]
@New0 Please review this and merge if you approve. Having this and #2692 in 1.8.0 will make working on admin refactor easier.

**Comment by New0** [comment]
@Shelob9 I didn't find PHP unit tests for the new routes so I started testing manually. 

- In classes/api/settings.php file, the 'permission_callback' declared for the new routes is not written (l23, l36, l56)
  - I tried to edite class declaration l12 to 'extends Caldera_Forms_API_CRUD' instead of 'implements  Caldera_Forms_API_Route' and used 'get_items_permissions_check' but had result {"code":"rest_forbidden","message":"Sorry, you are not allowed to do that.","data":{"status":401}}

File: classes/api/settings.php [hunk 1]
```diff
@@ -120,12 +120,50 @@ public function get_settings(){
...
+	* @since 1.8.0
+	*
+	* @param WP_REST_Request $request
+	*
+	* @return bool
+	*/
+	public function get_items_permissions_check( WP_REST_Request $request ){
+
+		//Check if a user is already set
+		if( wp_get_current_user()->ID === 0 ) {
```
Comment (Shelob9) [inline classes/api/settings.php]: @New0 Please remove this, like we discussed.

File: classes/api/settings.php [hunk 2]
```diff
@@ -120,12 +120,50 @@ public function get_settings(){
...
+	 *
+	 * @param WP_REST_Request $request
 	 *
 	 * @return bool
 	 */
-	public function permissions_check(){
+	public function create_item_permissions_check( WP_REST_Request $request ){
 
 		//Check if a user is already set
 		if( wp_get_current_user()->ID === 0 ) {
```
Comment (Shelob9) [inline classes/api/settings.php]: @New0 @New0 Please remove this, like we discussed.

**Pull request merged**